boards/DCM/Src/App/App_StartSwitch.c

```diff
@@ -0,0 +1,53 @@
+#include <assert.h>
+#include <stdbool.h>
+#include <stdlib.h>
+
+struct StartSwitch
+{
+    bool initial_switch_position;
+    bool switch_toggled;
+};
+
+struct StartSwitch *App_StartSwitch_Create()
+{
+    struct StartSwitch *start_switch = malloc(sizeof(struct StartSwitch));
+    assert(start_switch != NULL);
+
+    start_switch->initial_switch_position = false;
+    start_switch->switch_toggled          = false;
+    return start_switch;
+}
+
+void App_StartSwitch_SetInitialSwitchPosition(
+    struct StartSwitch *start_switch,
+    bool                position)
+{
+    start_switch->initial_switch_position = position;
+}
+
+void App_StartSwitch_SetSwitchToggled(
+    struct StartSwitch *start_switch,
+    bool                state)
+{
+    if (!state)
```

Comment:
To better represent the purpose of the function, I would check `if(start_switch->initial_switch_position == ON && state == OFF)`